### PR TITLE
Fix brave_sync API deps

### PR DESCRIPTION
**Pull request by darkdh**
fix https://github.com/brave/brave-browser/issues/1964

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [ ] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [ ] macOS
  - [x] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by simonhong** [comment]
nit: header define should be `BRAVE_BROWSER_EXTENSIONS_API_BRAVE_SYNC_EVENT_ROUTER_H` and
`BRAVE_BROWSER_EXTENSIONS_API_BRAVE_SYNC_API_H`

File: browser/extensions/BUILD.gn [hunk 1]
```diff
@@ -8,6 +8,10 @@ source_set("extensions") {
     "api/brave_rewards_api.h",
     "api/brave_shields_api.cc",
     "api/brave_shields_api.h",
+    "api/brave_sync_api.cc",
+    "api/brave_sync_api.h",
+    "api/brave_sync_event_router.cc",
```
Comment (simonhong) [inline browser/extensions/BUILD.gn]: I think we need to find more better place for router.cc in the future because it's not extension api.

**Comment by AlexeyBarabash** [comment]
master: https://github.com/brave/brave-core/commit/12032094c8fe1aa34d06daeea1a3836a66d706f9 
0.57.x: https://github.com/brave/brave-core/commit/80fa4479be0d9deccd8d83381ba8d4aa3b44f041
0.58.x:  https://github.com/brave/brave-core/commit/7f1250a5b98b7bcd44795186e2197dfd628f2d4b

**Comment by bbondy** [comment]
Pls also set the associated issue's milestone. I did it this time though.

**Pull request merged**